napari/layers/image/image.py

```diff
@@ -844,50 +796,8 @@ def _update_slice_response(self, response: _ImageSliceResponse) -> None:
 
         self._loaded = True
 
-    def _load_slice(self, data: ImageSliceData):
-        """Load the image and maybe thumbnail source. Currently used
-        for both sync and async slicing.
-
-        Parameters
-        ----------
-        data : ImageSliceData
-            Image slice
-        """
-        self._slice.load(data)
-        self._on_data_loaded(data)
-
-    def _on_data_loaded(self, data: ImageSliceData) -> None:
-        """The given data was loaded, use it now.
-
-        This routine is called for both sync and async slicing
-
-        Parameters
-        ----------
-        data : ImageSliceData
-            Image slice
-        """
-        # Transpose after the load.
-        data.transpose(self._get_order())
-
-        # The following block is still needed for both sync and async
-        # Pass the loaded data to the slice.
-        if not self._slice.on_loaded(data):
-            # Slice rejected it, was it for the wrong indices?
-            return
-
-        # Notify the world.
-        if self.multiscale:
-            self.events.scale()
-            self.events.translate()
-
-        # Announcing we are in the loaded state will make our node visible
-        # if it was invisible during the load.
-        # This is only needed for sync slicing
-        self.events.loaded()
-
```

Comment:
I'm very surprised that removing this works as expected. I had removed some things from `on_data_loaded` because they were moved out to `QtViewer`, but what remained here were things that I thought were needed because they weren't being called elsewhere (and it didn't work without them). How were you able to eliminate them?

Reply:
Any things in particular? Note that

- `transpose` was moved into slicing itself
- I think the `scale` and `translate` events are actually bugs since slicing can only change the `tile_to_data` transform, which is independent of those
- for the old-style sync path, I don't think `ImageSlice.load` and `ImageSlice.on_loaded` do much interesting other than apply `_ImageBase._raw_to_displayed`, which is now handled specifically for `Labels` which is the only thing that cares about it